src/python-common/ceph/deployment/service_spec.py

```diff
@@ -414,7 +414,10 @@ def __init__(self,
 
         assert service_type in ServiceSpec.KNOWN_SERVICE_TYPES, service_type
         self.service_type = service_type
-        self.service_id = service_id
+        if self.service_type in ['mon', 'mgr']:
```

Comment:
alertmanager, crash, grafana, node-exporter, prometheus and rbd-mirror as well. What about a new constant `ServiceSpec.SERVICE_HAS_SERVICE_ID`?

Reply:
I would suggest `ServiceSpec.SERVICE_REQUIRES_SERVICE_ID` as a more accurate (less misleading) name